app/src/main/java/org/sil/gatherwords/MainActivity.java

```diff
@@ -36,10 +47,22 @@ public void onCreateSessionClick(View v) {
 
     // TODO: Switch to CursorAdapter
     private class SessionListAdapter extends BaseAdapter {
+        private List<Session> sessions;
+        DatabaseAccess databaseAccess;
+
+        SessionListAdapter() {
+            try {
+                databaseAccess = new DatabaseAccess(ad);
+                sessions = (List<Session>) databaseAccess.select("session").get();
+
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
+            }
+        }
 
         @Override
         public int getCount() {
-            return 1;
+            return sessions.size();
```

Comment:
If we hit line 59, this will throw a npe.

Reply:
fixed in 90370be